src/gdb/gdb.c

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright (c) 2018, Intel Corporation
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *   * Neither the name of the Intel Corporation nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * Author: Marcin Rajwa <marcin.rajwa@linux.intel.com>
+ *
+ * GDB Stub - parse incoming GDB packets, control FW accordingly and provide
+ * a reply to the GDB server.
+ *
+ */
+
+#include <sof/gdb/gdb.h>
+#include <sof/gdb/ringbuffer.h>
+#include <arch/gdb/xtensa-defs.h>
+#include <string.h>
+#include <signal.h>
+#include <stdint.h>
+#include <xtensa/xtruntime.h>
+#include <sof/interrupt.h>
+
+/* main buffers */
+static unsigned char remcomInBuffer[BUFMAX];
+static unsigned char remcomOutBuffer[BUFMAX];
+
+/* local functions */
+static int getHex(unsigned char ch);
+static int hexToInt(unsigned char **ptr, int *intValue);
+static void putPacket(unsigned char *buffer);
+static void parseRequest(void);
+static unsigned char *getPacket(void);
+static void gdbLogException(char *message);
+
+/* registers backup conatiners */
+int sregs[256];
+int aregs[64];
```

Comment:
The naming here looks xtensa specific, should these also be "unsigned long" to align with target register size.

Reply:
Our target registers are 4 bytes, and both int and long are 4 bytes so it is all fine.

Reply:
@mrajwa Should the magic numbers like 64 be replaced with XCHAL_NUM_AREGS etc. defined by the arch spec?

Reply:
@mmaka1, yes I will replace it in following commits.